src/transactions/transaction.gateway.ts

```diff
@@ -1,6 +1,38 @@
 import { Injectable } from '@nestjs/common';
 import { Database } from 'src/system/database';
+import { ConfigProvider } from 'src/system/ConfigProvider';
 
 @Injectable()
 export class TransactionGateway extends Database {
+	transactionTable: string;
+	bankAccountTable: string;
+	constructor(configProvider: ConfigProvider) {
+		super(configProvider);
+		this.transactionTable = 'transactions';
+		this.bankAccountTable = 'bank_accounts';
+	}
+	getLastTransactions(days: number, userId: number): Promise<any> {
+		const sql = `
+		SELECT * FROM ${this.transactionTable} as tr
+		LEFT JOIN ${this.bankAccountTable} as bank
+		ON tr.account_id = bank.id
+		WHERE  (tr.date_time BETWEEN NOW() - INTERVAL ? DAY AND NOW()) AND (bank.user_id = ?);
```

Comment:
you don't need () around bank.user_id. also i think it should be tr.date_time > NOW() - interval ? day. it does the same think but looks cleaner